ganttproject/src/main/java/net/sourceforge/ganttproject/chart/TaskRendererImpl2.java

```diff
@@ -18,195 +18,164 @@
+    public int getVerticalOffset() {
+      return myModel.getVerticalOffset();
     }
+
     @Override
-    public boolean isMilestone(Task task) {
-      return ((TaskImpl)task).isLegacyMilestone();
+    public OffsetList getTasksUnitOffsets() {
+      return getChartModel().getDefaultUnitOffsets();
     }
+
     @Override
-    public boolean hasNestedTasks(Task task) {
-      return getChartModel().getTaskManager().getTaskHierarchy().hasNestedTasks(task);
+    public TimeUnit getProgressBarTimeUnit() {
+      return getChartModel().getTimeUnitStack().getDefaultTimeUnit();
     }
+
     @Override
-    public Color getColor(Task task) {
-      return task.getColor();
+    public net.sourceforge.ganttproject.chart.gantt.VerticalPartitioning getVerticalPartitioning() {
+      TaskContainmentHierarchyFacade containment = myModel.getTaskManager().getTaskHierarchy();
+      Map<ITaskSceneTask, Task> tasksMap = mapTaskSceneTask2Task(containment.getTasksInDocumentOrder(), myModel);
+      return new net.sourceforge.ganttproject.chart.gantt.VerticalPartitioning(
+        getVisibleTaskSceneTasks(),
+        (ITaskSceneTask t1, ITaskSceneTask t2) -> containment.areUnrelated(tasksMap.get(t1), tasksMap.get(t2))
+      );
     }
+
     @Override
-    public ShapePaint getShapePaint(Task task) {
-      if (task.getShape() == null) {
-        return ShapeConstants.TRANSPARENT;
-      }
-      return task.getShape();
+    public List<ITask> getVisibleTasks() {
+      return ImmutableList.copyOf(
+        DependencySceneApiAdapterKt.tasks2itasks(myModel.getVisibleTasks()).values()
+      );
     }
+
     @Override
-    public boolean hasNotes(Task task) {
-      return !Strings.isNullOrEmpty(task.getNotes());
+    public List<ITaskSceneTask> getVisibleTaskSceneTasks() {
+      return ImmutableList.copyOf(
+        mapTaskSceneTask2Task(TaskRendererImpl2.this.getVisibleTasks(), myModel).keySet()
+      );
     }
-  };
 
-  class TaskActivityChartApi implements TaskActivitySceneBuilder.ChartApi {
-    TaskActivityChartApi() {
-    }
     @Override
-    public Date getChartStartDate() {
-      return myModel.getOffsetAnchorDate();
+    public List<ITaskSceneTask> getTasksInDocumentOrder() {
+      TaskContainmentHierarchyFacade containment = myModel.getTaskManager().getTaskHierarchy();;
+      return ImmutableList.copyOf(
+        mapTaskSceneTask2Task(containment.getTasksInDocumentOrder(), myModel).keySet()
+      );
     }
+
     @Override
-    public Date getEndDate() {
-      return getChartModel().getEndDate();
+    public List<GanttPreviousStateTask> getBaseline() {
+      return myModel.getBaseline();
     }
+
     @Override
-    public OffsetList getBottomUnitOffsets() {
-      return getChartModel().getBottomUnitOffsets();
+    public TaskActivitySceneBuilder.ChartApi getChartApi(TaskLabelSceneBuilder<ITaskSceneTask> labelsRenderer) {
+      return new TaskActivitySceneChartApi(myModel) {
+        @Override
+        public int getRowHeight() {
```

Comment:
it looks as if it is the same code as in GanttChartSceneBuilder::calculateRowHeight